src/Redux_Todos/TodoList.js

```diff
@@ -0,0 +1,75 @@
+import React, { useState } from "react";
+import { connect } from "react-redux";
+import { addTodo, deleteTodo, toggleTodo } from "./redux/actions";
+import InputComponent from "./InputComponent";
+
+const TodoList = ({ todos, addTodo, deleteTodo, toggleTodo }) => {
+  const [inputValue, setInputValue] = useState("");
+
+  const handleInputChange = (e) => {
+    setInputValue(e.target.value);
+  };
+
+  const handleAddTodo = () => {
+    if (inputValue) {
+      addTodo(inputValue);
+      setInputValue("");
+    }
+  };
+
+  const handleDeleteTodo = (id) => {
+    deleteTodo(id);
+  };
+
+  const handleToggleTodo = (id) => {
+    toggleTodo(id);
+  };
+
+  return (
+    <div className="todo-list">
+      <div>
+        {todos.map((todo) => (
+          <div className="todo" key={todo.id}>
+            <span
+            onClick={() => handleToggleTodo(todo.id)}
+            style={{
+                textDecoration: todo.isCompleted ? 'line-through' : 'none'
+            }}
```

Comment:
In real projects try to avoid inline styles, and prefer CSS
Also, check the video about the performance in telegram chat, there are explanations why such an approach is bad practice
```
className={`todo-text ${todo.isCompleted ? "todo-completed" : ""}`}
```